README.md

```diff
@@ -74,8 +74,59 @@ This command will return the following output:
+[15:19:30] [INFO] testing 'PostgreSQL AND error-based - WHERE or HAVING clause'
+[15:19:30] [INFO] testing 'Microsoft SQL Server/Sybase AND error-based - WHERE or HAVING clause (IN)'
+[15:19:30] [INFO] testing 'Oracle AND error-based - WHERE or HAVING clause (XMLType)'
+[15:19:30] [INFO] testing 'Generic inline queries'
+[15:19:30] [INFO] testing 'PostgreSQL > 8.1 stacked queries (comment)'
+[15:19:30] [INFO] testing 'Microsoft SQL Server/Sybase stacked queries (comment)'
+[15:19:30] [INFO] testing 'Oracle stacked queries (DBMS_PIPE.RECEIVE_MESSAGE - comment)'
+[15:19:30] [INFO] testing 'MySQL >= 5.0.12 AND time-based blind (query SLEEP)'
+[15:19:30] [INFO] testing 'PostgreSQL > 8.1 AND time-based blind'
+[15:19:30] [INFO] testing 'Microsoft SQL Server/Sybase time-based blind (IF)'
+[15:19:30] [INFO] testing 'Oracle AND time-based blind'
+```
+
+But returned that `all tested parameters do not appear to be injectable`. There are a few possible reasons for that. 
+
+- **SQLMap's Testing Strategy**: SQLMap follows a set of predefined testing strategies and payloads to detect SQL injection vulnerabilities. It doesn't necessarily cover all possible variations of SQL injection and may not detect vulnerabilities if they deviate from its testing patterns.
+
+- **Test Parameters**: SQLMap relies on the parameters provided in the URL or form data for testing. If the parameters are not easily identifiable as injection points, SQLMap may not detect them as potential vulnerabilities.
+
+- **Custom Code**: SQLMap is a general-purpose tool that may not fully understand the specifics of the `app.py`. It might not recognize the SQL injection vulnerability in the Flask application because it's looking for more common patterns or vulnerabilities.
+
+- **Mitigation Measures**: In the original prompt `sqlmap -u "http://0.0.0.0:8080/users/admin" -v`, I am using SQLAlchemy and parameterized queries to mitigate SQL injection vulnerabilities. These measures are effective in preventing SQL injection attacks. SQLMap may not detect these vulnerabilities if it cannot find any exploitable SQL injection points.
+
+- **False Negatives**: False negatives (where SQLMap doesn't detect a vulnerability) can happen. SQLMap might require specific circumstances or payloads to trigger an injection, and if those conditions are not met during its automated testing, it may report no vulnerabilities.
+
+It's important to note that even if SQLMap doesn't detect a vulnerability, it doesn't necessarily mean the code is secure. Manual code review and thorough testing are essential for identifying and addressing potential security issues in your application. While parameterized queries and ORM libraries like SQLAlchemy help prevent SQL injection, other security considerations, such as input validation, authentication, and authorization, should also be part of a security strategy.
```

Comment:
This is a great note! And a worthy mention and risk of purely testing using automated tools.